src/System.AppContext/src/System/AppContext.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using Windows.ApplicationModel;
+
+namespace System
+{
+    public static class AppContext
+    {
+        [Flags]
+        private enum SwitchValueState
+        {
+            HasFalseValue = 0x1,
+            HasTrueValue = 0x2,
+            HasLookedForOverride = 0x4, // Not used on .NET Native
+            UnknownValue = 0x8 // Not used on .NET Native
+        }
+
+        private static Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
+        private static readonly object s_syncLock = new object();
+
+        public static string BaseDirectory
+        {
+            get
+            {
+                return Package.Current.InstalledLocation.Path;
+            }
+        }
+
+        /// <summary>
+        /// Try to get the value of the switch.
+        /// </summary>
+        /// <param name="switchName">The name of the switch</param>
+        /// <param name="isEnabled">A variable where to place the value of the switch</param>
+        /// <returns>A return value of true represents that the switch was set and <paramref name="isEnabled"/> contains the value of the switch</returns>
+        public static bool TryGetSwitch(string switchName, out bool isEnabled)
+        {
+            if (switchName == null)
+                throw new ArgumentNullException("switchName");
+            if (switchName.Length == 0)
+                throw new ArgumentException(SR.Argument_EmptyName, "switchName");
+
+            // By default, the switch is not enabled.
+            isEnabled = false;
```

Comment:
Good point! I missed that!

Do you think this is going to safe a lot of cycles?